metacat-common-server/src/main/java/com/netflix/metacat/common/server/usermetadata/AuthorizationService.java

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package com.netflix.metacat.common.server.usermetadata;
+
+import com.netflix.metacat.common.QualifiedName;
+
+/**
+ * Authorization Service API.
+ *
+ * @author zhenl
+ * @since 1.2.0
+ */
+public interface AuthorizationService {
+    /**
+     * Check metacat acl property if this operation is permitted.
+     *
+     * @param userName      username
+     * @param name           qualified Name
+     * @param op             operation
+     * @return true if isUnauthorized
+     */
+    default boolean isUnauthorized(final String userName,
```

Comment:
This will work but I would prefer this method to be a void method. 
void authorize(final String username, final QualifiedName name, final Operation op). This will check the access privilege and if the user does not have the access throw an generic error. <username> cannot perform <op> on <name>.